pkg/copy/fileinfo.go

```diff
@@ -83,20 +83,20 @@ func ResolveHostPath(path string) (*FileInfo, error) {
 // is preserved.  The filepath API among tends to clean up a bit too much but
 // we *must* preserve this data by all means.
 func PreserveBasePath(original, resolved string) string {
-	// Handle "/"
-	if strings.HasSuffix(original, "/") {
-		if !strings.HasSuffix(resolved, "/") {
-			resolved += "/"
+	// Handle "/" or "\\" on Windows.
+	if strings.HasSuffix(original, string(os.PathSeparator)) {
```

Comment:
If this file is compiled on Windows, the check here should use `os.IsPathSeparator` (to check for both `/` and `\\`).

Reply:
There’s a couple places in the PR where I need to update to use that instead of directly comparing to `os.PathSeparator`.

Reply:
Actually, the other change I was thinking of was in the copier package in buildah, but the usage there is fine since it's a path that should only ever be reached on non-Windows systems.

Reply:
I refactored `copy.PreserveBasePath` based on `archive.PreserveTrailingDotOrSeparator` from [containers/storage](https://github.com/containers/storage/blob/67591acf91ff3321a8a13e962654c23ea34c4b31/pkg/archive/copy.go#L30) since that has a cleaner Windows friendly approach to the same behavior.